src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/CreateNextDataSetVersionFunctionTests.cs

```diff
@@ -0,0 +1,499 @@
+            var dataSetVersionImport = Assert.Single(nextDataSetVersion.Imports);
+            Assert.Equal(nextDataSetVersion.Id, dataSetVersionImport.DataSetVersionId);
+            Assert.NotEqual(Guid.Empty, dataSetVersionImport.InstanceId);
+            Assert.Equal(DataSetVersionImportStage.Pending, dataSetVersionImport.Stage);
+
+            // Assert the response view model values match the created data set version and import.
+            Assert.Equal(updatedDataSet.Id, responseViewModel.DataSetId);
+            Assert.Equal(nextDataSetVersion.Id, responseViewModel.DataSetVersionId);
+            Assert.Equal(dataSetVersionImport.InstanceId, responseViewModel.InstanceId);
+
+            // Assert the processing orchestrator was scheduled with the correct arguments
+            Assert.NotNull(processNextDataSetVersionContext);
+            Assert.NotNull(startOrchestrationOptions);
+            Assert.Equal(new ProcessDataSetVersionContext { DataSetVersionId = nextDataSetVersion.Id },
+                processNextDataSetVersionContext);
+            Assert.Equal(new StartOrchestrationOptions { InstanceId = dataSetVersionImport.InstanceId.ToString() },
+                startOrchestrationOptions);
+        }
+
+        [Fact]
+        public async Task ReleaseFileIdIsEmpty_ReturnsValidationProblem()
+        {
+            var result = await CreateNextDataSetVersion(
+                dataSetId: Guid.NewGuid(),
+                releaseFileId: Guid.Empty);
+
+            var validationProblem = result.AssertBadRequestWithValidationProblem();
+
+            validationProblem.AssertHasNotEmptyError(
+                nameof(NextDataSetVersionCreateRequest.ReleaseFileId).ToLowerFirst());
+        }
+
+        [Fact]
+        public async Task DataSetIdIsEmpty_ReturnsValidationProblem()
+        {
+            var result = await CreateNextDataSetVersion(
+                dataSetId: Guid.Empty,
+                releaseFileId: Guid.NewGuid());
+
+            var validationProblem = result.AssertBadRequestWithValidationProblem();
+
+            validationProblem.AssertHasNotEmptyError(
+                nameof(NextDataSetVersionCreateRequest.DataSetId).ToLowerFirst());
+        }
+
+        [Fact]
+        public async Task DataSetIdIsNotFound_ReturnsValidationProblem()
+        {
+            var subjectId = Guid.NewGuid();
+            
+            var (releaseFile, releaseMetaFile) = DataFixture.DefaultReleaseFile()
+                .WithReleaseVersion(DataFixture.DefaultReleaseVersion()
+                    .WithApprovalStatus(ReleaseApprovalStatus.Draft))
+                .WithFiles([
+                    DataFixture
+                        .DefaultFile(FileType.Data)
+                        .WithSubjectId(subjectId),
+                    DataFixture
+                        .DefaultFile(FileType.Metadata)
+                        .WithSubjectId(subjectId)
+                ])
+                .GenerateList()
+                .ToTuple2();
```

Comment:
Could maybe use your `AddDataAndMetadataFiles` method here to shorten?

Reply:
Oh yeah!  Thanks, changed that